app/index.js

```diff
@@ -166,105 +165,108 @@ module.exports = yeoman.generators.Base.extend({
     }
   },
 
-  makeConfig: function () {
-    this.sourceRoot(path.join(__dirname, this.options['template-path']));
-
-    this.templateArray = function (files, comments, coffee) {
-      var str = [];
-      _.each(comments, function (comment) {
-        str.push('\n      ' + (coffee ? '# ' : '// ') + comment);
-      });
-      _.uniq(files).forEach(function (item, index) {
-        str.push('\n      \'' + item + '\'');
-        if (index + 1 !== files.length) {
-          if (!coffee) {
-            str.push(',');
-          }
-        }
-      });
-      str.push('\n    ');
-      return str.join('');
-    };
-
-    this.template(
-      this.options['config-file'],
-      path.join(this.options['config-path'], this.options['config-file'])
-    );
-  },
-
-  writeDependencies: function() {
-
-    this.options.plugins.push('grunt-karma');
-    if (this.options.coffee) {
-      this.options.plugins.push('coffee-script');
-    }
-
-    var data = this.fs.readJSON(this.destinationPath('package.json'));
-    if(!data) {
-      this.log.error('Could not open package.json for reading.');
-      return;
-    }
-
-    data.devDependencies = data.devDependencies || {};
-      this.options.plugins.forEach(function (plugin) {
-        data.devDependencies[plugin] = '*';
-      })
-      data.devDependencies = sortedObject(data.devDependencies);
-
-    this.fs.writeJSON(this.destinationPath('package.json'), data);
-  },
-
-  writeGruntFile: function () {
-    if (!this.options['gruntfile-path']) {
-      return;
-    }
-
-    this.gruntfile.loadNpmTasks('grunt-karma');
-    this.gruntfile.insertConfig(
-      'karma',
-      JSON.stringify({
-        unit: {
-          options: {
-            'autoWatch': false,
-            'browsers': this.options.browsers,
-            'configFile': [
-              this.options['config-path'],
-              '/',
-              this.options['config-file']
-            ].join(''),
-            'singleRun': true
+  writing: {
+    makeConfig: function () {
+      this.sourceRoot(path.join(__dirname, this.options['template-path']));
+
+      this.templateArray = function (files, comments, coffee) {
+        var str = [];
+        _.each(comments, function (comment) {
+          str.push('\n      ' + (coffee ? '# ' : '// ') + comment);
+        });
+        _.uniq(files).forEach(function (item, index) {
+          str.push('\n      \'' + item + '\'');
+          if (index + 1 !== files.length) {
+            if (!coffee) {
+              str.push(',');
+            }
           }
-        }
-      })
-    );
-    this.gruntfile.registerTask('test', ['karma']);
-  },
+        });
+        str.push('\n    ');
+        return str.join('');
+      };
+
+      this.fs.copyTpl(
+        this.templatePath(this.options['config-file']),
+        this.destinationPath(path.join(this.options['config-path'], this.options['config-file'])),
+        this
```

Comment:
Please, let's move away from this horrible hack of using this as a template context.